### DISCUSSION TITLE
slint vs iced

### DISCUSSION THREAD

**Discussion by soloturn** [slint-ui/slint#2224] [General] [answered]
could somebody point to a dicsussion slint vs iced, or give a couple of notes why both do exist and what the advantage is?

**Comment by ogoffart** [comment, marked as answer]
I can't have a unbiased opinion, but here are some points in which Slint does better:

- **Tooling, and dev experience**: We have a LSP server that can be used from many [editors](https://github.com/slint-ui/slint/blob/master/editors/README.md). It provides language support for the slint macro, and most importantly, a live-preview which enable fast designing of UI.
- **Desktop Native looking UI**: We aim to create native looking UI on desktop platforms. This doesn't same to be the use case for iced, where even the button doesn't look like a native button at all.  Note that if you don't want native looking widget with Slint, it is also possible (and easy) to create your own custom looking widgets.
- **Low-end embedded**: We aim to make it easy to create UI for embedded devices.  It scales from devices with OpenGL running Linux, to micro-controllers in `#![no_std]` with only hundreds of kB of RAM.  Iced cannot run on such devices

Now, iced might have advantages. Here is what i've been told:

 - Slint uses a Markup language, while Iced is "pure" Rust. I don't see why that's an advantage for Iced. Learning the Slint markup language is not more difficult than learning the API for libraries. Also even Iced use some macros which have their own syntax. Rust syntax is often a bit too verbose for the need of the UI, and is also incompatible with live-preview. 
 - Iced is in a more permissive license which allow companies to take the code without contributing anything back. Slint use a dual license model with a GPL license for open source software, and a proprietary license for proprietary software. Note that the commercial license is free of charge as long as you attribute Slint (disclose that you are using Slint), which should work for most users that are not big corp that can afford a paid license anyway.
